app/services/form1010cg/service.rb

```diff
@@ -3,25 +3,85 @@
 # This service manages the interactions between CaregiversAssistanceClaim, CARMA, and Form1010cg::Submission.
 module Form1010cg
   class Service
-    def submit_claim!(claim_data)
-      claim = SavedClaim::CaregiversAssistanceClaim.new(claim_data)
+    attr_reader :claim
+
+    NOT_FOUND = 'NOT_FOUND'
+
+    def initialize(claim)
       claim.valid? || raise(Common::Exceptions::ValidationErrors, claim)
 
-      carma_submission = CARMA::Models::Submission.from_claim(claim)
+      @claim = claim
+      @cached_icns = {}
+    end
 
-      carma_submission.submit!
+    def process_claim!
+      assert_veteran_status
+
+      carma_submission = CARMA::Models::Submission.from_claim(claim, build_metadata).submit!
 
       Form1010cg::Submission.new(
         carma_case_id: carma_submission.carma_case_id,
         submitted_at: carma_submission.submitted_at
       )
     end
 
+    def assert_veteran_status
+      raise_unprocessable if icn_for('veteran') == NOT_FOUND
+    end
+
+    def build_metadata
+      claim.form_subjects.each_with_object({}) do |namespace, acc|
```

Comment:
Not sure what acc means here -- and is there a word other than "namespace" that describes what you're iterating through? Even if its a `form_subject` it might be clearer.

Reply:
Oops. I used `acc` (accumulator) out of habit (it's a common practice for this same function in javascript `Array.prototype.reduce`).

 I think `metadata` would make more sense here. I will also use `form_subject` instead of `namespace`. Thanks!